tla/ccfraft.tla

```diff
@@ -121,98 +132,248 @@ VARIABLE configurations
 \* TODO: re-add a server in a raft_scenario test?
 VARIABLE removedFromConfiguration
 
-reconfigurationVars == <<reconfigurationCount, removedFromConfiguration, configurations>>
+RemovedFromConfigurationTypeInv ==
+    removedFromConfiguration \subseteq Servers
+
+reconfigurationVars == <<
+    reconfigurationCount, 
+    removedFromConfiguration, 
+    configurations
+>>
+
+ReconfigurationVarsTypeInv ==
+    /\ ReconfigurationCountTypeInv
+    /\ ConfigurationsTypeInv
+    /\ RemovedFromConfigurationTypeInv
 
 \* A set representing requests and responses sent from one server
 \* to another. With CCF, we have message integrity and can ensure unique messages.
 \* Messages only records messages that are currently in-flight, actions should
 \* removed messages once received.
 VARIABLE messages
 
+\* Helper function for checking the type safety of log entries
+EntryTypeOK(entry) ==
+    /\ entry.term \in Nat \ {0}
+    /\ \/ /\ entry.contentType = TypeEntry
+          /\ entry.value \in Nat \ {0}
+       \/ /\ entry.contentType = TypeSignature
+          /\ entry.value = Nil
+       \/ /\ entry.contentType = TypeReconfiguration
+          /\ entry.value \subseteq Servers
+
+AppendEntriesRequestTypeOK(m) ==
+    /\ m.mtype = AppendEntriesRequest
```

Comment:
If we're doing some spring cleaning, can we get rid of all the (seemingly unnecessary) "m" prefixes?

Reply:
Sure, I don't see why not. See 3b39e58097cc32d5ba0b0fb68ff14bfe2f827ccf